libcloudforensics/providers/gcp/internal/storage.py

```diff
@@ -263,12 +274,57 @@ def CreateBucket(
       Dict[str, Any]: An API operation object for a Google Cloud Storage bucket.
            https://cloud.google.com/storage/docs/json_api/v1/buckets#resource
     """
+    if bucket.startswith('gs://'):
+      bucket = bucket[5:]
     gcs_buckets = self.GcsApi().buckets()
     body = {'name': bucket, 'labels': labels}
     request = gcs_buckets.insert(
         project=self.project_id,
         predefinedAcl=predefined_acl,
         predefinedDefaultObjectAcl=predefined_default_object_acl,
         body=body)
-    response = request.execute()  # type: Dict[str, Any]
+    try:
+      response = request.execute()  # type: Dict[str, Any]
+    except HttpError as exception:
+      if exception.resp.status == 409:
+        raise errors.ResourceCreationError(
+            'Bucket {0:s} already exists: {1!s}'.format(bucket, exception),
+            __name__) from exception
+      raise errors.ResourceCreationError(
+          'Unknown error occurred when creating bucket:'
+          ' {0!s}'.format(exception), __name__) from exception
     return response
+
+  def GetObject(self,
+                gcs_path: str,
+                out_file: Optional[str] = None) -> str:
+    """Gets the contents of an object in a Google Cloud Storage bucket.
+
+    Args:
+      gcs_path (str): Full path to the object (ie: gs://bucket/dir1/dir2/obj)
+      out_file (str): Path to the local file that will be written.
+        If not provided, will create a temporary file.
+    Returns:
+      str: The filename of the written object.
+    """
+    if not gcs_path.startswith('gs://'):
+      gcs_path = 'gs://' + gcs_path
+    gcs_objects = self.GcsApi().objects()
+    (bucket, filename) = SplitStoragePath(gcs_path)
+    request = gcs_objects.get_media(bucket=bucket, object=filename)
+
+    if not out_file:
+      outputdir = tempfile.mkdtemp()
+      logger.info('Created temporary directory {0:s}'.format(outputdir))
+      out_file = os.path.join(outputdir, os.path.basename(filename))
+
+    outputfile = open(out_file, 'wb')
+    downloader = googleapiclient.http.MediaIoBaseDownload(outputfile, request)
+
+    done = False
+    while not done:
```

Comment:
Is there any scenario where `downloader.next_chunk()` could hang e.g. due to network outage and make this an infinite loop?

Reply:
This is the logic as given in the documentation. The underlying class uses a timeout (default 60 seconds)